build_system.sh

```diff
@@ -67,10 +71,6 @@ exec_as_root() {
   docker exec $MOUNT_CONTAINER_ID "$@"
 }
 
-# Create host user in container (fixes namespace.so error)
-USERNAME=$(whoami)
```

Comment:
you need to replace missing $USERNAME in `docker exec -u $USERNAME $MOUNT_CONTAINER_ID "$@"`

Reply:
Yes! Good catch. It actually works as is (at least when using zsh) because USERNAME is set by default. But better to explicitly set it. I'll take over your code.